pxtblocks/blocklyloader.ts

```diff
@@ -3199,4 +3212,18 @@ namespace pxt.blocks {
             }
         }
     }
+
+    let iconCanvasCache: Map<string> = {};
+    function iconToFieldImage(id: string): Blockly.FieldImage {
+        let url = iconCanvasCache[id];
```

Comment:
check that URL exists

Reply:
What if it doesn't exist? We need to return a Field image.

Reply:
We should emit an error in the console and skip that image or put a default image